### PR TITLE
Py3 fixes

### PR DESCRIPTION
**Pull request by miguelsousa**
I've switched off AppVeyor's pr event for the moment.

### CODE REVIEW

File: afdko/Tools/SharedData/FDKScripts/pdfgen.py [hunk 1]
```diff
@@ -529,7 +536,7 @@ def setDash(self, array=[], phase=0):
             self._code.append('[%s %s] 0 d' % (array, phase))
```
Comment (cjchapman) [inline afdko/Tools/SharedData/FDKScripts/pdfgen.py]: please change line 533 above to:
```python
    def setDash(self, array=None, phase=0):
```
and then add code to handle the `None` case

reference: http://docs.python-guide.org/en/latest/writing/gotchas/

File: afdko/Tools/SharedData/FDKScripts/pdfgen.py [hunk 2]
```diff
@@ -682,48 +689,48 @@ def drawInlineImage(self, image, x,y, width=None,height=None):
...
+                if x[0] != 8:
+                    raise PDFError('JPEG must have 8 bits per component')
+                y = struct.unpack('BB', image.read(2))
+                height = (y[0] << 8) + y[1]
+                y = struct.unpack('BB', image.read(2))
+                width = (y[0] << 8) + y[1]
+                y = struct.unpack('B', image.read(1))
+                color = y[0]
+                return width, height, color
+                done = 1
```
Comment (cjchapman) [inline afdko/Tools/SharedData/FDKScripts/pdfgen.py]: this line (`done = 1`) is unreachable, so please remove it

File: afdko/Tools/SharedData/FDKScripts/pdfgen.py [hunk 3]
```diff
@@ -791,13 +798,13 @@ def setPageTransition(self, effectname=None, duration=1,
         try:
             args = PageTransitionEffects[effectname]
         except KeyError:
-            raise 'PDFError', 'Unknown Effect Name "%s"' % effectname
+            raise PDFError('Unknown Effect Name "%s"' % effectname)
             self._pageTransitionString = ''
```
Comment (cjchapman) [inline afdko/Tools/SharedData/FDKScripts/pdfgen.py]: lines 802 & 803 are unreachable, so please remove them

File: afdko/Tools/SharedData/FDKScripts/pdfmetrics.py [hunk 4]
```diff
@@ -3,34 +3,33 @@
...
 # 1
-import string
-
+from __future__ import print_function, absolute_import
 
 
 StandardEnglishFonts = [
-	'Courier', 'Courier-Bold', 'Courier-Oblique', 'Courier-BoldOblique',  
-	'Helvetica', 'Helvetica-Bold', 'Helvetica-Oblique', 
+	'Courier', 'Courier-Bold', 'Courier-Oblique', 'Courier-BoldOblique',
```
Comment (cjchapman) [inline afdko/Tools/SharedData/FDKScripts/pdfmetrics.py]: please replace the tabs in lines 31-35 with spaces

File: afdko/Tools/SharedData/FDKScripts/pdfmetrics.py [hunk 5]
```diff
@@ -3,34 +3,33 @@
...
+	'Helvetica', 'Helvetica-Bold', 'Helvetica-Oblique',
 	'Helvetica-BoldOblique',
 	'Times-Roman', 'Times-Bold', 'Times-Italic', 'Times-BoldItalic',
 	'Symbol','ZapfDingbats']
```
Comment (cjchapman) [inline afdko/Tools/SharedData/FDKScripts/pdfmetrics.py]: please replace the tabs in line 39 below with spaces

File: afdko/Tools/SharedData/FDKScripts/pdfmetrics.py [hunk 6]
```diff
@@ -3,34 +3,33 @@
...
+to sort out the fontname case issue and the resolution of PIDDLE fonts to
 Postscript font names within this module, but have not yet done so.
 
 
 13th June 1999
 """
 # 1
-import string
-
+from __future__ import print_function, absolute_import
```
Comment (cjchapman) [inline afdko/Tools/SharedData/FDKScripts/pdfmetrics.py]: need to add:
```python
import os
```
(see comment below)

File: afdko/Tools/SharedData/FDKScripts/pdfmetrics.py [hunk 7]
```diff
@@ -91,10 +91,10 @@ def __init__(self):
         global widths
         self.__widtharrays = widths
 
-        
+
     def loadfont(self, fontname):
         filename = AFMDIR + os.sep + fontname + '.afm'
```
Comment (cjchapman) [inline afdko/Tools/SharedData/FDKScripts/pdfmetrics.py]: `AFMDIR` is undefined
(also, need to import `os`)

File: afdko/Tools/SharedData/FDKScripts/pdfmetrics.py [hunk 8]
```diff
@@ -91,10 +91,10 @@ def __init__(self):
...
-        print 'cache loading',filename
+        print('cache loading', filename)
         assert os.path.exists(filename)
```
Comment (cjchapman) [inline afdko/Tools/SharedData/FDKScripts/pdfmetrics.py]: (another reason to import `os`)

File: afdko/Tools/SharedData/FDKScripts/pdfutils.py [hunk 9]
```diff
@@ -1,9 +1,13 @@
...
 # 1
+from __future__ import print_function, absolute_import
+
 import os
-import string
-import cStringIO
+try:
+    from StringIO import StringIO
+except ImportError:
+    from io import StringIO
```
Comment (cjchapman) [inline afdko/Tools/SharedData/FDKScripts/pdfutils.py]: need to add:
```python
import glob
```
(see note below)

File: afdko/Tools/SharedData/FDKScripts/pdfutils.py [hunk 10]
```diff
@@ -61,10 +65,10 @@ def preProcessImages(spec):
```
Comment (cjchapman) [inline afdko/Tools/SharedData/FDKScripts/pdfutils.py]: please change line 61 from:
```python
    if type(spec) is types.StringType:
```
to:
```python
    if isinstance(spec, types.StringType):
```
(also, need to import `glob` for line 62 above)

File: afdko/Tools/SharedData/FDKScripts/otfPDF.py [hunk 11]
```diff
@@ -53,7 +53,7 @@ def _curveToOne(self, pt1, pt2, pt3):
 	def _closePath(self):
 		self.numPaths += 1
 		self.noPath = 1
-		curPath = self.pathList[-1] 
+		curPath = self.pathList[-1]
```
[the remark this answers is not in the document]
Comment (miguelsousa) [reply afdko/Tools/SharedData/FDKScripts/otfPDF.py]: I've removed `curPath` in a previous commit, but it made the **digiplot** test fail. There seems to be some sort of recursiveness in this class.

[the remark this answers is not in the document]
Comment (cjchapman) [reply afdko/Tools/SharedData/FDKScripts/otfPDF.py]: I just tried commenting out the `curPath` line from `_closePath()` and the **digiplot** test passed on my machine. If you can reproduce the failure, I'd like to look over your shoulder to see what's going on. Thanks.

File: afdko/Tools/SharedData/FDKScripts/pdfmetrics.py [hunk 12]
```diff
@@ -56,26 +322,26 @@ def parseAFMfile(filename):
...
+
+        _, cid = chunks[0].split()
+        _, width = chunks[1].split()
+        # n, name = chunks[2].split()
+        # b, x1, y1, x2, y2 = chunks[3].split()
+        widths[int(cid)] = int(width)
+
     # by default, any empties should get the width of a space
     for i in range(len(widths)):
         if widths[i] == 0:
```
[the remark this answers is not in the document]
Comment (miguelsousa) [reply afdko/Tools/SharedData/FDKScripts/pdfmetrics.py]: done

**Pull request merged**